src/apps/the-exchange/components/EnterAmount/EnterAmount.tsx

```diff
@@ -29,72 +62,155 @@ const EnterAmount = ({ type, tokenSymbol }: EnterAmountProps) => {
+                    addressesEqual(balance.token, assetAddress)
+                );
+            });
+
+        const assetBalanceValue = assetBalance ? assetBalance.balance : '0';
+        const balance = processEth(assetBalanceValue, swapToken.decimals ?? 18);
+
+        // Check if the value exceeds the max token amount limit
+        if (balance && tokenAmount > balance) {
+            setErrorMessage(`The max amount limit in your wallet is ${balance.toFixed(4)} ${swapToken?.symbol}`);
+        }
+    };
 
+    // getOffer will be called every time the swap amount or the swap/receive token is changed
     useEffect(() => {
-        if (type === SwapReceive.SWAP) {
-            setInputValue(amountSwap ? amountSwap.tokenAmount.toString() : '0.00');
-        } else {
-            setInputValue(amountReceive ? amountReceive.tokenAmount.toString() : '0.00');
+        setInputValue(amountSwap ? amountSwap.tokenAmount : 0);
+        if (amountSwap?.tokenAmount) {
+            setIsOfferLoading(true);
+            debouncedGetOffer();
+            tokenBalanceLimit(amountSwap?.tokenAmount);
         }
-    }, [type, amountSwap, amountReceive]);
+        // Clean-up debounce on component unmount
+        return () => {
+            debouncedGetOffer.cancel();
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [amountSwap, swapToken, receiveToken]);
 
+    // When the token amount value changes, the input value reflects that change
+    // the handleTokenAmountChange will make sure that we get a USD price, and will then add it to amountSwap
     const handleTokenAmountChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
         const value = e.target.value;
-        setInputValue(value);
-    
-        if (type === SwapReceive.SWAP && swapToken) {
+        setInputValue(Number(value));
+        setErrorMessage('');
+
+        // Might need to find a way to debounce the getPrice call too
+        if (type === CardPosition.SWAP && swapToken) {
             const usdPrice = await getPrice(swapToken.address, swapToken.chainId);
             if (usdPrice) {
                 setAmountSwap({ tokenAmount: Number(value), usdAmount: usdPrice.usd * Number(value) });
             } else {
                 setAmountSwap({ tokenAmount: Number(value), usdAmount: 0 });
             }
-        } else if (type === SwapReceive.RECEIVE && receiveToken) {
-            const usdPrice = await getPrice(receiveToken.address, receiveToken.chainId);
-            if (usdPrice) {
-                setAmountReceive({ tokenAmount: Number(value), usdAmount: usdPrice.usd * Number(value) });
-            } else {
-                setAmountReceive({ tokenAmount: Number(value), usdAmount: 0 });
-            }
         }
     };
 
+    // Function to render offer based on loading and offer state
+    const renderOffer = () => {
+        if (isOfferLoading) {
+            return <CircularProgress size={36} sx={{ color: '#343434' }} />;
```

Comment:
Could we pull `#343434` from Tailwind or something from the theme?

Reply:
I usually would, but because this is a material ui component, to change the color, and apply a tailwind themed color, I would need to create a theme folder for material ui, and impose the theme to this component. While when using sx={{}}, I can quickly just change the color, but cannot use the tailwind color in that case

Reply:
Understood!